pkg/chunked/storage_linux.go

```diff
@@ -1324,6 +1332,35 @@ func (c *chunkedDiffer) findAndCopyFile(dirfd int, r *internal.FileMetadata, cop
 	return false, nil
 }
 
+func makeEntriesFlat(mergedEntries []internal.FileMetadata) ([]internal.FileMetadata, error) {
+	var new []internal.FileMetadata
+
+	hashes := make(map[string]string)
+	for i := range mergedEntries {
+		if mergedEntries[i].Type != TypeReg {
+			continue
+		}
+		if mergedEntries[i].Digest == "" {
+			continue
```

Comment:
Should missing digests maybe be an error

Reply:
the files without digest are the empty files and they are simply ignored from the output.  I can improve the check though

Reply:
Well, You might have for whatever reason a buggy json file with non-empty files without digests. Might be good to report that.